### ISSUE TITLE
Matomo tag manager "preview/debugging" window not working with Content Security Policy

### ISSUE THREAD

**Issue by MattiasRuljeff** [matomo-org/tag-manager#583]
Everything works with the Matomo tag manager, except for the preview/debugging window when I use "Content Security Policy (CSP)", it gets blocked by the Cross-Origin Resource Sharing "CORS".  The window works fine if I disable the CPS...
Any help with the issue would be very appreciated!

**Comment by AltamashShaikh** [comment]
hey @MattiasRuljeff is there any URL where we can see the above behaviour ?

**Comment by MattiasRuljeff** [comment]
https://hk-fw-test.azurewebsites.net/?mtmPreviewMode=rifOUcD8&mtmSetDebugFlag=1

**Comment by snake14** [comment]
Thank you for providing that URL @MattiasRuljeff . It look like you're trying to load the Matomo Tag Manager code as part of your bundle.js file? When I tried loading [bundle.js](https://hk-fw-test.azurewebsites.net/dist/client/bundle.js?v=7bd09fa5c73864a413cf), there was an error in the JavaScript console and it looks like you might have multiple CSP violations. You might need to do a little more fine tuning in your CSP directive.
![image](https://user-images.githubusercontent.com/11394099/207957653-07cf3fd4-5181-448e-84a0-3c7c041251e0.png)

**Comment by MattiasRuljeff** [comment]
I will take a look at it! Thanks for the help so far!

**Comment by AltamashShaikh** [comment]
@MattiasRuljeff Should we close this issue or do you need more help ?

**Comment by MattiasRuljeff** [comment]
I have not got the time to check the issue yet, but you can close the issue for now and if the error still remains I´ll open a new issue!
